### PR TITLE
Modify Rule S6329: Make generic and add GCP

### PR DESCRIPTION
**Pull request by hendrik-buchwald-sonarsource**
Implementation ticket: https://jira.sonarsource.com/browse/SONARIAC-325

### CODE REVIEW

**Comment by hendrik-buchwald-sonarsource** [comment]
@loris-s-sonarsource I have only added Compute instance for now. Do you think that is sufficient or should I add all the other possible resources as well?

File: rules/S6329/terraform/rule.adoc [hunk 1]
```diff
@@ -6,30 +6,54 @@ include::../recommended.adoc[]
...
 
+GCP Compute instance that has a public IP address assigned to it:
+
+----
+resource "google_compute_instance" "example" {
+  network_interface {
+    network = "default"
+
+    access_config {  // Sensitive
+      // Ephemeral public IP
```
Comment (loris-s-sonarsource) [inline rules/S6329/terraform/rule.adoc:29]:
```suggestion
    access_config {  # Sensitive
      # Ephemeral public IP
```

**Review by loris-s-sonarsource — CHANGES_REQUESTED** [review]
Oh no, I started a review, but there is already a PR open for this: https://github.com/SonarSource/rspec/pull/655

**Comment by loris-s-sonarsource** [comment]
> @loris-s-sonarsource I have only added Compute instance for now, as an example. Do you think that is sufficient or should I add all the other possible resources as well?

@hendrik-buchwald-sonarsource Yes, I think that GCP doesn't seem to have resources with a clear public IP management (except maybe [notebooks_instance#no_public_ip](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/notebooks_instance#no_public_ip))

**Comment by hendrik-buchwald-sonarsource** [comment]
@loris-s-sonarsource ah, I will rebase this tomorrow on your changes. I have some resources with public IP, see the security-expected-issues PR. I checked more resources but discarded them for some reason or another.

**Review by loris-s-sonarsource — APPROVED** [review]
LGTM! Minor suggestion on snippet comments

**Pull request merged**